base/deprecated.jl

```diff
@@ -1417,6 +1417,22 @@ function LibGit2.set_remote_url(path::AbstractString, url::AbstractString; remot
     LibGit2.set_remote_url(path, remote, url)
 end
 
+# NOTE: rename the function replace_new from strings/util.jl to replace
+# when this replace is removed from deprecated.jl
+function replace(s::AbstractString, pat, f, n::Integer)
+    if n <= 0
+        depwarn(string("`replace(s, pat, r, count)` with `count <= 0` is deprecated, use ",
+                       "`replace(s, pat, r, typemax(Int))` or `replace(s, pat, r)` instead"),
+                :replace)
+        replace(s, pat, f)
+    else
+        Base.replace_new(String(s), pat, f, clamp(n, 0, typemax(Int)) % Int)
```

Comment:
`Base.` shouldn't be needed here either, deprecated.jl is included within the Base module